src/python/pants/bsp/spec/base.py

```diff
@@ -252,3 +253,14 @@ class StatusCode(IntEnum):
 
     # Execution was cancelled.
     CANCELLED = 3
+
+
+class BSPData:
+    """Mix-in for BSP spec types that can live in a data field."""
+
+    @classproperty
+    def DATA_KIND(cls):
```

Comment:
Should probably be marked abstract?

Reply:
I thought we had a preference to avoid ABCMeta?